src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java

```diff
@@ -37,17 +68,292 @@ public WorkOrderBuildDecoration(final String structureName, final String workOrd
+            if (canBuild(citizen))
+            {
+                sendMessage = false;
+            }
+
+            if (!job.hasWorkOrder() && canBuild(citizen))
+            {
+                job.setWorkOrder(this);
+                this.setClaimedBy(citizen);
+                return;
+            }
+        }
+
+        sendBuilderMessage(colony, hasBuilder, sendMessage);
+    }
+
+    /**
+     * send a message from the builder.
+     *
+     * Suppressing Sonar Rule squid:S1172
+     * This rule does "Unused method parameters should be removed"
+     * But in this case extending class may need to use the sendMessage parameter
+     * @param colony which the work order belong to
+     * @param hasBuilder true if we have a builder for this work order
+     * @param sendMessage true if we need to send the message
+     */
+    @SuppressWarnings("squid:S1172")
+    protected void sendBuilderMessage(@NotNull final Colony colony, final boolean hasBuilder, final boolean sendMessage)
+    {
+        if (hasSentMessageForThisWorkOrder)
+        {
+            return;
+        }
+
+        if (!hasBuilder)
+        {
+            hasSentMessageForThisWorkOrder = true;
+            LanguageHandler.sendPlayersMessage(colony.getMessageEntityPlayers(),
+              "entity.builder.messageNoBuilder");
+        }
     }
 
+    /**
+     * Checks if a builder may accept this workOrder.
+     *
+     * Suppressing Sonar Rule squid:S1172
+     * This rule does "Unused method parameters should be removed"
+     * But in this case extending class may need to use the citizen parameter
+     * @param citizen which could build it or not
+     * @return true if he is able to.
+     */
+    @SuppressWarnings("squid:S1172")
+    protected boolean canBuild(@NotNull final CitizenData citizen)
+    {
+        return true;
+    }
+
+
     @Override
     public boolean isValid(final Colony colony)
     {
         return true;
     }
 
+    @NotNull
+    @Override
+    protected WorkOrderType getType()
+    {
+        return WorkOrderType.BUILD;
+    }
+
+
     @Override
     protected String getValue()
     {
         return workOrderName;
     }
+
+    /**
+     * Returns the ID of the building (aka ChunkCoordinates).
+     *
+     * @return ID of the building.
+     */
+    public BlockPos getBuildingLocation()
+    {
+        return buildingLocation;
+    }
+
+    /**
+     * Get the name the structure for this work order.
+     *
+     * @return the internal string for this structure.
+     */
+    public String getStructureName()
+    {
+        return structureName;
+    }
+
+    /**
+     * Gets how many times this structure should be rotated.
+     *
+     * Suppressing Sonar Rule squid:S1172
+     * This rule does "Unused method parameters should be removed"
+     * But in this case extending class may need to use the world parameter
+     * @param world where the decoration is
+     * @return building rotation.
+     */
+    @SuppressWarnings("squid:S1172")
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S1172' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)